src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,95 @@
+import usersFromServer from '../../api/users';
+import { Todo } from '../../types/Todo';
+import React, { FC, useState, FormEvent } from 'react';
+import { findUserById } from '../../utils/FindUserById';
+
+interface Props {
+  onAdd: (todo: Todo) => void;
+  todos: Todo[];
+}
+
+export const TodoForm: FC<Props> = ({ onAdd, todos }) => {
+  const [newTitle, setNewTitle] = useState('');
+  const [newUserId, setNewUserId] = useState(0);
+
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+
+  function handleFormReset() {
+    setNewTitle('');
+    setNewUserId(0);
+    setTitleError('');
+    setUserError('');
+  }
+
+  function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    if (!newTitle) {
+      setTitleError('Please enter a title');
+    }
+
+    if (!newUserId) {
+      setUserError('Please choose a user');
+    }
+
+    if (newTitle && newUserId) {
+      const nextId: number = todos[todos.length - 1].id + 1;
+
+      const newTodo: Todo = {
+        id: nextId,
+        title: newTitle,
+        completed: false,
+        userId: newUserId,
+        user: findUserById(newUserId),
+      };
+
+      onAdd(newTodo);
+
+      handleFormReset();
+    }
+  }
+
+  return (
+    <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
+      <div className="field">
+        <input
+          type="text"
+          data-cy="titleInput"
+          value={newTitle}
+          placeholder="Some task"
+          onChange={event => {
+            setTitleError('');
```

Comment:
[CODE STYLE] - Don't use one-letter variable naming. It could be confusing for another developer, who reads your code. Also, it would really hard to find this variable in the code editor using search. Consider renaming `event` to `changeEvent` or `inputEvent`.